DQMOffline/Trigger/plugins/LepHTMonitor.cc

```diff
@@ -0,0 +1,488 @@
+  //Offline electron definition
+  bool isGood(const reco::GsfElectron &el, const reco::Vertex::Point &pv_position,
+              const reco::BeamSpot::Point &bs_position,
+              const edm::Handle<reco::ConversionCollection> &convs, bool pass_id,
+	      const double lep_counting_threshold, const double lep_iso_cut, const double lep_eta_cut,  
+	      const double d0_cut_b, const double dz_cut_b,  const double d0_cut_e,  const double dz_cut_e){
+
+    //Electron ID
+    if(!pass_id) return false;
+
+    //pT
+    if(el.pt()<lep_counting_threshold || fabs(el.superCluster()->eta()) > lep_eta_cut) return false;
+
+    //Isolation
+    const auto &iso = el.pfIsolationVariables();
+    const float absiso = iso.sumChargedHadronPt
+      + std::max(0.0, iso.sumNeutralHadronEt + iso.sumPhotonEt -0.5 * iso.sumPUPt);
+    const float relisowithdb = absiso/el.pt();
+    if(relisowithdb>lep_iso_cut) return false; 
+    
+    //Conversion matching
+    bool pass_conversion = false;
+    if(convs.isValid()){
+      pass_conversion = !ConversionTools::hasMatchedConversion(el, convs, bs_position);
+    }
+    else{
+      edm::LogError("LepHTMonitor") << "Electron conversion matching failed.\n";
+    }
+    if(!pass_conversion) return false;
+    
+
+    //Impact parameter
+    float d0 = 999., dz=999.;
+    if(el.gsfTrack().isNonnull()){
+      d0=-(el.gsfTrack()->dxy(pv_position));
+      dz = el.gsfTrack()->dz(pv_position);
+    }
+    else{
+      edm::LogError("LepHTMonitor") << "Could not read electron.gsfTrack().\n";
+      return false;
+    }
+    float etasc = el.superCluster()->eta();
+    if(fabs(etasc)>1.479){//Endcap 
+      if(fabs(d0)>d0_cut_e || fabs(dz)>dz_cut_e) return false;
+ 
+    }else{//Barrel
+      if(fabs(d0)>d0_cut_b || fabs(dz)>dz_cut_b) return false;
+    }
+
+    return true;
+  }
+
+  //Offline muon definition
+  bool isGood(const reco::Muon &mu, const reco::Vertex &pv,
+	      const double lep_counting_threshold, const double lep_iso_cut, const double lep_eta_cut,
+	      const double d0_cut, const double dz_cut, int muonIDlevel){
+
+    const reco::Vertex::Point &pv_position = pv.position();
+
+    //Muon pt and eta acceptance
+    if(mu.pt()<lep_counting_threshold || fabs(mu.eta())>lep_eta_cut) return false;
+    try{
+      //Muon isolation
+      const auto &iso = mu.pfIsolationR04();
+      const float absiso = iso.sumChargedHadronPt
+	+ std::max(0.0, iso.sumNeutralHadronEt + iso.sumPhotonEt -0.5 * iso.sumPUPt);
+      const float relisowithdb = absiso/mu.pt();
+      if(relisowithdb>lep_iso_cut) return false;
+    }catch(...){
+      edm::LogWarning("LepHTMonitor") << "Could not read muon isolation.\n";
+      return false;
+    }
+    try{
+      //Muon ID
+      bool pass_id = false;
+      if(muonIDlevel == 1) pass_id = muon::isLooseMuon(mu);
+      else if(muonIDlevel == 3) pass_id = muon::isTightMuon(mu,pv);
+      else pass_id = muon::isMediumMuon(mu); 
+
+      if(!pass_id) return false;
+
+    }catch(...){
```

Comment:
none of the called methods throws, as far as I can see from their code; I'll simply remove the `try` and the catch blocks.